include/common/types.h

```diff
@@ -599,10 +576,6 @@ template <> inline ValType ValTypeFromType<float>() noexcept {
 template <> inline ValType ValTypeFromType<double>() noexcept {
   return ValType(TypeCode::F64);
 }
-// wasm interface types
-template <> inline ValType ValTypeFromType<StrVariant>() noexcept {
-  return ValType(TypeCode::String);
-}
```

Comment:
TODO: should be
```c++
template <> inline ValType ValTypeFromType<std::string>() noexcept {
  return InterfaceType(...);
}
```

Reply:
besides that, we will need a concrete way to handle the case like
```
template<> inline ValType ValTypeFromType<std::vector<T>>() noexcept {
  return InterfaceType(
    TypeCode::List,
    ValTypeFromType<T>
  );
}
```

@q82419 Do you have any idea about such construction?

Reply:
```c++
template <typename T>
inline ValType ValTypeFromType<std::vector<T>>() noexcept {
  return InterfaceType(TypeCode::String, ValTypeFromType<T>());
}
```
is invalid, but above also invalid due to `T` is missing.